src/web/app/pages-admin/admin-accounts-page/admin-accounts-page.component.ts

```diff
@@ -67,14 +43,27 @@ export class AdminAccountsPageComponent implements OnInit {
    * Loads the account information based on the given ID.
    */
   loadAccountInfo(instructorid: string): void {
-    const paramMap: { [key: string]: string } = { instructorid };
-    this.httpRequestService.get('/accounts', paramMap).subscribe((resp: AccountInfo) => {
-      this.instructorCourses = resp.instructorCourses;
-      this.studentCourses = resp.studentCourses;
-      this.accountInfo = resp.accountInfo;
+    this.accountService.getAccount(instructorid).subscribe((resp: Account) => {
+      this.accountInfo = resp;
     }, (resp: ErrorMessageOutput) => {
       this.statusMessageService.showErrorMessage(resp.error.message);
     });
+
+    this.courseService.getStudentCoursesInMasqueradeMode(instructorid).subscribe((resp: Courses) => {
+      this.studentCourses = resp.courses;
+    }, (resp: ErrorMessageOutput) => {
+      if (resp.status !== 403) {
+        this.statusMessageService.showErrorMessage(resp.error.message);
+      }
+    });
+
+    this.courseService.getInstructorCoursesInMasqueradeMode(instructorid).subscribe((resp: Courses) => {
```

Comment:
As a result, the function `(resp: Courses) => {}` will be called two times. i.e. the instructorCourses will be overridden.